sfge/src/main/java/com/salesforce/collections/CollectionUtil.java

```diff
@@ -186,5 +181,20 @@ public static <T, U> List<U> newImmutableListOf(
                 .collect(Collectors.collectingAndThen(Collectors.toList(), ImmutableList::copyOf));
     }
 
+    /**
+     * Peek into the stack to get the last element. Converts value into an Optional instead of
+     * throwing an {@link EmptyStackException} when stack is empty.
+     *
+     * @param stack to look at
+     * @param <T> Stack's generic
+     * @return last element added to the stack
+     */
+    public static <T> Optional<T> peek(Stack<T> stack) {
```

Comment:
Moving this here since the logic is needed on a couple of different Stacks and would be helpful to reuse in future as well.